oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/query/DurationUtils.java

```diff
@@ -81,33 +81,25 @@ public long endTimeDurationToSecondTimeBucket(Step step, String dateStr) {
     }
 
     public long startTimeToTimestamp(Step step, String dateStr) throws ParseException {
-        switch (step) {
-            case MONTH:
-                return new SimpleDateFormat("yyyy-MM").parse(dateStr).getTime();
-            case DAY:
-                return new SimpleDateFormat("yyyy-MM-dd").parse(dateStr).getTime();
-            case HOUR:
-                return new SimpleDateFormat("yyyy-MM-dd HH").parse(dateStr).getTime();
-            case MINUTE:
-                return new SimpleDateFormat("yyyy-MM-dd HHmm").parse(dateStr).getTime();
-            case SECOND:
-                return new SimpleDateFormat("yyyy-MM-dd HHmmss").parse(dateStr).getTime();
-        }
-        throw new UnexpectedException("Unsupported step " + step.name());
+        return dateStrToTimeStamp(step, dateStr, 0);
     }
 
     public long endTimeToTimestamp(Step step, String dateStr) throws ParseException {
+        return dateStrToTimeStamp(step, dateStr, 1);
+    }
+
+    private long dateStrToTimeStamp(Step step, String dateStr, int plus) throws ParseException {
         switch (step) {
             case MONTH:
-                return new DateTime(new SimpleDateFormat("yyyy-MM").parse(dateStr)).plusMonths(1).getMillis();
+                return new DateTime(new SimpleDateFormat("yyyy-MM").parse(dateStr)).plusMonths(plus).getMillis();
```

Comment:
`plus 0` looks like strange. And codes are unclear.

Reply:
Okay, I will revert it to master